src/Data/Functor/Foldable/TH.hs

```diff
@@ -70,12 +74,79 @@ import Paths_recursion_schemes (version)
 -- as we don't try to do better than
 -- <https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/glasgow_exts.html#deriving-functor-instances GHC's DeriveFunctor>.
 --
-makeBaseFunctor :: Name -> DecsQ
-makeBaseFunctor = makeBaseFunctorWith baseRules
+-- For trickier data-types, like rose-tree (see also 'Cofree')
+--
+-- @
+-- data Rose f a = Rose a (f (Rose f a))
+-- @
+--
+-- we can invoke 'makeBaseFunctor' with instance declaration,
+-- to provide needed context for instances. (c.f. @StandaloneDeriving@)
+--
+-- @
+-- 'makeBaseFunctor' [d| instance Functor f => Recursive (Rose f a) |]
```

Comment:
It would be worth writing out the generated code here to emphasize that the `Functor f` context is carried over.